File: benchmarks/fixtures/angular_fixture/src/app/services/api.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../environments/environment';

@Injectable({ providedIn: 'root' })
export class ApiService {
  private apiUrl = environment.apiUrl;

  constructor(private http: HttpClient) {}

  getUsers() {
    return this.http.get('/api/users');
  }

  createUser(data: any) {
    return this.http.post('/api/users', data);
  }

  getPost(id: string) {
    return this.http.get(`/api/posts/${id}`);
  }

  deletePost(id: string) {
    return this.http.delete(`/api/posts/${id}`);
  }

  updateSettings(settings: any) {
    return this.http.put('/api/settings', settings);
  }
}

File: benchmarks/fixtures/android_native_fixture/app/src/main/java/com/example/demo/data/ApiService.kt
package com.example.demo.data

import retrofit2.http.*

interface ApiService {
    @GET("/api/users")
    suspend fun getUsers(): List<User>

    @POST("/api/users")
    suspend fun createUser(@Body user: User): User

    @GET("/api/users/{id}")
    suspend fun getUser(@Path("id") id: String): User

    @DELETE("/api/users/{id}")
    suspend fun deleteUser(@Path("id") id: String)

    @GET("/api/posts")
    suspend fun getPosts(): List<Post>

    @POST("/api/posts")
    suspend fun createPost(@Body post: Post): Post
}

companion object {
    const val BASE_URL = "https://api.example.com/v1"
}
